src/Detector/Go/GoDetector.cs

```diff
@@ -0,0 +1,99 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Oryx.Detector.Go
+{
+    /// <summary>
+    /// An implementation of <see cref="IPlatformDetector"/> which detects Go applications.
+    /// </summary>
+    class GoDetector : IGoPlatformDetector
+    {
+        private readonly ILogger<GoDetector> _logger;
+        private readonly DetectorOptions _options;
+
+        /// <summary>
+        /// Creates an instance of <see cref="GoDetector"/>.
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger{GoDetector}"/>.</param>
+        /// <param name="options">The <see cref="DetectorOptions"/>.</param>
+        public GoDetector(ILogger<GoDetector> logger, IOptions<DetectorOptions> options)
+        {
+            _logger = logger;
+            _options = options.Value;
+        }
+
+        public PlatformDetectorResult Detect(DetectorContext context)
+        {
+            string appDirectory = string.Empty;
+            var sourceRepo = context.SourceRepo;
+            // check if go.mod exists
+            if (!sourceRepo.FileExists(GoConstants.GoDotModFileName))
+            {
+                _logger.LogDebug(
+                    $"Could not find {GoConstants.GoDotModFileName} in repo");
+                return null;
+            }
+            _logger.LogInformation($"Found {GoConstants.GoDotModFileName} at the root of the repo. ");
+
+            // TODO: check if go.sum or /cmd/repo_name/*.go 
+            var version = GetVersion(context);
```

Comment:
looking at https://github.com/GoogleCloudPlatform/golang-samples/blob/master/go.mod ... i feel go.mod contains the information of go version and the other version of direct dependencies where as go.sum contains versions of the both direct and indrect packages and the chuksum validations.

Reply:
Correct, I've removed TODO